Build failure with GCC 15

Arch Linux x86_64 with GHC 9.4.8:

```
Preprocessing library for system-fileio-0.3.16.6..
Building library for system-fileio-0.3.16.6..
[1 of 1] Compiling Filesystem       ( lib/Filesystem.hs, dist/build/Filesystem.dyn_o )

lib/Filesystem.hs:79:48: warning: [-Wunused-imports]
    The import of ‘when’ from module ‘Control.Monad’ is redundant
   |
79 | import           Control.Monad (forM_, unless, when)
   |                                                ^^^^

lib/Filesystem.hs:95:1: warning: [-Wunused-imports]
    The import of ‘System.IO.Error’ is redundant
      except perhaps to import instances from ‘System.IO.Error’
    To import instances alone, use: import System.IO.Error()
   |
95 | import           System.IO.Error (IOError)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

lib/Filesystem.hs:925:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘throwErrnoPathIfMinus1’
    |
925 | throwErrnoPathIfMinus1 loc path = CError.throwErrnoPathIfMinus1 loc (encodeString path)
    | ^^^^^^^^^^^^^^^^^^^^^^
In file included from /usr/include/dirent.h:25,

                 from lib/hssystemfileio-unix.c:9:0: error:

/usr/include/features.h:199:3: error:
     warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
      199 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
          |   ^~~~~~~
    |
199 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
    |   ^

lib/hssystemfileio-unix.c:19:1: error:
     error: conflicting types for ‘hssystemfileio_alloc_dirent’; have ‘struct dirent *(void *)’
       19 | hssystemfileio_alloc_dirent(void *void_dir)
          | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
   |
19 | hssystemfileio_alloc_dirent(void *void_dir)
   | ^

In file included from lib/hssystemfileio-unix.c:1:0: error:

lib/hssystemfileio-unix.h:7:1: error:
     note: previous declaration of ‘hssystemfileio_alloc_dirent’ with type ‘struct dirent *(void)’
        7 | hssystemfileio_alloc_dirent();
          | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  |
7 | hssystemfileio_alloc_dirent();
  | ^
lib/hssystemfileio-unix.c: In function ‘hssystemfileio_readdir’:

lib/hssystemfileio-unix.c:57:17: error:
     warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
       57 |                 int rc = readdir_r(dir, dirent, &dirent_result);
          |                 ^~~
   |
57 |                 int rc = readdir_r(dir, dirent, &dirent_result);
   |                 ^

/usr/include/dirent.h:185:12: error:
     note: declared here
      185 | extern int readdir_r (DIR *__restrict __dirp,
          |            ^~~~~~~~~
    |
185 | extern int readdir_r (DIR *__restrict __dirp,
    |            ^
`gcc' failed in phase `C Compiler'. (Exit code: 1)
```

--- a/system-fileio/lib/hssystemfileio-unix.h
+++ b/system-fileio/lib/hssystemfileio-unix.h
@@ -4,7 +4,7 @@
 struct dirent;
 
 struct dirent *
-hssystemfileio_alloc_dirent();
+hssystemfileio_alloc_dirent(void *void_dir);
 
 void
 hssystemfileio_free_dirent(struct dirent *);

